### PR TITLE
Global search: created by and edited by support multiple ids

### PR DESCRIPTION
**Pull request by qnkhuat**
Update search api so that we can search for multiple creators, multiple last editors.

The URL patterns should be sth like : `api/search?creator_id=1&creator_id=2`

### CODE REVIEW

File: src/metabase/api/search.clj [hunk 1]
```diff
@@ -457,17 +457,17 @@
...
    last_edited_at [:maybe ms/PositiveInt]
-   last_edited_by [:maybe ms/PositiveInt]
+   last_edited_by [:maybe [:or ms/PositiveInt [:sequential ms/PositiveInt]]]
    verified       [:maybe true?]}
   (query-model-set (search-context {:search-string  q
                                     :archived       archived
                                     :table-db-id    table-db-id
                                     :created-at     created_at
-                                    :created-by     created_by
+                                    :created-by     (u/one-or-many created_by)
```
Comment (calherries) [inline src/metabase/api/search.clj:468]: Why isn't this `(set (u/one-or-many created_by))`? Then you wouldn't need to modify `created_at` again in `search-context`. It feels unnecessary to separate the "preparation" of this value in two steps

File: test/metabase/api/search_test.clj [hunk 2]
```diff
@@ -796,12 +796,14 @@
...
     (with-search-items-in-root-collection search-term
-      (t2.with-temp/with-temp
+      (mt/with-temp
         [:model/User      {user-id :id}      {:first_name "Explorer" :last_name "Curious"}
+         :model/User      {user-id-2 :id}    {:first_name "Explorer" :last_name "Hubble"}
          :model/Card      {card-id :id}      {:name (format "%s Card 1" search-term) :creator_id user-id}
          :model/Card      {card-id-2 :id}    {:name (format "%s Card 2" search-term) :creator_id user-id
                                               :collection_id (:id (collection/user->personal-collection user-id))}
          :model/Card      {card-id-3 :id}    {:name (format "%s Card 3" search-term) :creator_id user-id :archived true}
+         :model/Card      {card-id-4 :id}      {:name (format "%s Card 4" search-term) :creator_id user-id-2}
```
Comment (calherries) [inline test/metabase/api/search_test.clj:806]: Indentation
```suggestion
         :model/Card      {card-id-4 :id}    {:name (format "%s Card 4" search-term) :creator_id user-id-2}
```

**Review by calherries — APPROVED** [review]
LGTM

File: src/metabase/api/search.clj (hunk 1 above)
Comment (qnkhuat) [reply src/metabase/api/search.clj]: fair, I've made the change in [`b7aa764` (#34302)](https://github.com/metabase/metabase/pull/34302/commits/b7aa76480bd953de5e4ef6fcb01255b6f2b4de0d)

**Pull request merged**